src/java/org/apache/cassandra/index/sai/plan/VectorTopKProcessor.java

```diff
@@ -89,6 +104,31 @@ public VectorTopKProcessor(ReadCommand command)
         this.limit = command.limits().count();
     }
 
+    /**
+     * Stage to use for parallel index reads.
+     * Defined by -DIndexReadStage.cassandra.stage=<value> where value matches one of the Stage enum values.
+     * INDEX_READ is default, READ is a possibility, or IMMEDIATE to disable parallel reads.
+     * Others are not recommended.
+     * @see org.apache.cassandra.concurrent.Stage
+     *
+     * INDEX_READ uses 2 * cpus threads by default but can be overridden with -DIndexReadStage.cassandra.threads=<value>
+     *
+     * @return stage to use, default INDEX_READ
+     */
+    private static LocalAwareExecutorService getExecutor()
+    {
+        boolean isParallel = Boolean.parseBoolean(System.getProperty("IndexRead.cassandra.parallel", "true"));
```

Comment:
pls move this into CassandraRelevantProperties and follow naming convention there